sql/policy_internal.sql

```diff
@@ -19,163 +19,125 @@ AS '@MODULE_PATHNAME@', 'ts_policy_refresh_cagg_proc'
 LANGUAGE C;
 
 CREATE OR REPLACE PROCEDURE
-_timescaledb_internal.policy_compression_interval( job_id INTEGER, 
-   htid INTEGER,
-   lag INTERVAL,
-   maxchunks INTEGER,
-   verbose_log BOOLEAN,
-   recompress_enabled BOOLEAN)
+_timescaledb_internal.policy_compression_execute(
+  job_id              INTEGER,
+  htid                INTEGER,
+  lag                 ANYELEMENT,
+  maxchunks           INTEGER,
+  verbose_log         BOOLEAN,
+  recompress_enabled  BOOLEAN)
 AS $$
 DECLARE
-  htoid regclass;
-  chunk_rec record;
-  numchunks integer := 1;
+  htoid       REGCLASS;
+  chunk_rec   RECORD;
+  numchunks   INTEGER := 1;
 BEGIN
 
-  SELECT format('%I.%I',schema_name, table_name) INTO htoid
+  SELECT format('%I.%I', schema_name, table_name) INTO htoid
   FROM _timescaledb_catalog.hypertable
   WHERE id = htid;
 
-  FOR chunk_rec IN
-    SELECT show.oid, ch.schema_name, ch.table_name, ch.status
-    FROM show_chunks( htoid, older_than => lag) as show(oid)
-      INNER JOIN pg_class pgc ON pgc.oid = show.oid
-      INNER JOIN pg_namespace pgns ON pgc.relnamespace = pgns.oid
-      INNER JOIN _timescaledb_catalog.chunk ch ON ch.table_name = pgc.relname and ch.schema_name = pgns.nspname and ch.hypertable_id = htid
-    WHERE ch.dropped is false and  (ch.status = 0 OR ch.status = 3)
-  LOOP
-    IF chunk_rec.status = 0 THEN
-       PERFORM compress_chunk( chunk_rec.oid );
-    ELSIF chunk_rec.status = 3 AND recompress_enabled = 'true' THEN
-       PERFORM recompress_chunk( chunk_rec.oid );
-    END IF;
-    COMMIT;
-    IF verbose_log THEN
-       RAISE LOG 'job % completed processing chunk %.%', job_id, chunk_rec.schema_name, chunk_rec.table_name;
-    END IF;
-    numchunks := numchunks + 1;
-    IF maxchunks > 0 AND numchunks >= maxchunks THEN  
-         EXIT; 
-    END IF;  
-  END LOOP;
-END;
-$$ LANGUAGE PLPGSQL;
-
-CREATE OR REPLACE PROCEDURE
-_timescaledb_internal.policy_compression_integer( job_id INTEGER, 
-   htid INTEGER,
-   lag BIGINT,
-   maxchunks INTEGER,
-   verbose_log BOOLEAN,
-   recompress_enabled BOOLEAN)
-AS $$
-DECLARE
-  htoid regclass;
-  chunk_rec record;
-  numchunks integer := 0;
-  lag_integer BIGINT;
-BEGIN
-
-  SELECT format('%I.%I',schema_name, table_name) INTO htoid
-  FROM _timescaledb_catalog.hypertable
-  WHERE id = htid;
-
-  --for the integer case , we have to compute the lag w.r.t 
+  -- for the integer cases, we have to compute the lag w.r.t
   -- the integer_now function and then pass on to show_chunks
-  lag_integer := _timescaledb_internal.subtract_integer_from_now( htoid, lag);
+  IF pg_typeof(lag) IN ('BIGINT'::regtype, 'INTEGER'::regtype, 'SMALLINT'::regtype) THEN
```

Comment:
```suggestion
  IF pg_typeof(lag) IN ('BIGINT', 'INTEGER', 'SMALLINT') THEN
```

Reply:
It doesn't work because `pg_typeof` function returns `REGTYPE`.

```sql
fabrizio=# \df pg_typeof
                           List of functions
   Schema   |   Name    | Result data type | Argument data types | Type 
------------+-----------+------------------+---------------------+------
 pg_catalog | pg_typeof | regtype          | "any"               | func
(1 row)

fabrizio=# SELECT pg_typeof('BIGINT'::regtype) IN ('BIGINT', 'INTEGER', 'SMALLINT');
 ?column? 
----------
 f
(1 row)
```

Reply:
And without explicit type cast to `REGTYPE` it will be recognized as `unknown`

```sql
fabrizio=# SELECT pg_typeof('BIGINT'), pg_typeof('BIGINT'::regtype);
 pg_typeof | pg_typeof 
-----------+-----------
 unknown   | regtype
(1 row)
```